frontend/src/Judge/reducers.js

```diff
@@ -78,6 +90,19 @@ const queue = (state = initialQueueState, action) => {
 const queues = (state = {}, action) => {
   // Proxy all queue actions to the particular queue we want to target
   switch (action.type) {
+    case actions.FETCH_SUBMISSIONS:
+      if (action.payload.submissions.length === 0) {
+        return state
+      }
+      // proxy the action to the subqueue
+      const showId = action.payload.submissions[0].show.id
+      return {
+        [showId]: queue(
+          state[showId],
+          action
+        ),
+        ...state
```

Comment:
One last nit: I believe this needs to go before, otherwise you could be overwriting the `[showId]` object you just made:

```js
return {
  ...state,
  [showId]: queue(
    state[showId],
    action
  )
}       
```